Code/GraphMol/SmilesParse/SmilesParse.h

```diff
@@ -32,8 +47,30 @@ struct RDKIT_SMILESPARSE_EXPORT SmilesParserParams {
   bool skipCleanup =
       false; /**<  skip the final cleanup stage (for internal use) */
 };
-RDKIT_SMILESPARSE_EXPORT RWMol *SmilesToMol(const std::string &smi,
-                                            const SmilesParserParams &params);
+
+struct RDKIT_SMILESPARSE_EXPORT SmartsParserParams {
+  int debugParse = 0; /**< enable debugging in the SMARTS parser*/
+  std::map<std::string, std::string> *replacements =
```

Comment:
Does replacements have to be a nullptr?  Could it be an empty map?  This would handle lifetimes better at the expense of a copy when setting replacements.

Reply:
Could be an empty map, but this is the same point as above with the recommendation from the core guidelines.

Also Re: lifetimes. I don't see that argument. If we say that a raw pointer in the new API never transfers ownership (and I very much intend that this will be the case) then this goes away.

Reply:
If the struct can’t be copied I’m in complete agreement.  Otherwise it is too easy to use a deleted pointer.

Reply:
We can use `std::optional` introduced in C++17.

Reply:
I replaced the pointer with a map in the v2 API. We can't change the v1 API (which has had the pointer since more or less the beginning)

Reply:
> We can use `std::optional` introduced in C++17.

That's a good point, but it feels like overkill in this situation?